sanic/app.py

```diff
@@ -690,6 +698,16 @@ def url_for(self, view_name: str, **kwargs):
 
         # parse the remainder of the keyword arguments into a querystring
         query_string = urlencode(kwargs, doseq=True) if kwargs else ""
+
+        if external and port:
+            port = int(port)
```

Comment:
I know this may sound silly, but shouldn't we wrap this call with a `try/except` in case `int` raises a `ValueError` or `TypeError`? Just in case someone tries to pass some random string or some other weird object to the `_port` argument or `SERVER_PORT` configuration key? At least we will prevent any issues in the future with this ...

Reply:
yes, sounds reasonable.

Reply:
Great, thanks @lixxu !

Reply:
I actually like this more as it was, makes it impossible to use it wrong. the try/except logic lets you pass a nonsense port and doesn't even complain